File: test/fixtures/zig/src/person.zig
const std = @import("std");

/// A person with a name and age.
pub const Person = struct {
    name: []const u8,
    age: u32,

    pub fn init(name: []const u8, age: u32) Person {
        return Person{ .name = name, .age = age };
    }

    pub fn greet(self: Person, allocator: std.mem.Allocator) ![]u8 {
        return std.fmt.allocPrint(allocator, "Hello, {s}", .{self.name});
    }
};

File: test/fixtures/csharp/Person.cs
namespace CSharpFixture;

/// <summary>A person with a name and age.</summary>
public class Person
{
    public string Name { get; set; }
    public int Age { get; set; }

    public Person(string name, int age)
    {
        Name = name;
        Age = age;
    }

    public string Greet()
    {
        return $"Hello, {Name}";
    }
}
